package/kernel/mac80211/patches/981-rt2x00-partially-revert-AGC-init.patch

```diff
@@ -0,0 +1,16 @@
+--- a/drivers/net/wireless/ralink/rt2x00/rt2800lib.c
++++ b/drivers/net/wireless/ralink/rt2x00/rt2800lib.c
+@@ -4109,12 +4109,7 @@ static void rt2800_config_channel(struct
+ 		rt2800_bbp_write(rt2x00dev, 196, reg);
+ 
+ 		/* AGC init */
+-		if (rt2x00_rt(rt2x00dev, RT6352))
+-			reg = 0x04;
```

Comment:
All vendor drivers I've seen got
```
R66 = 0x04 + 2 * GET_LNA_GAIN(pAd);
```
hence I doubt reverting to `0x1c` instead of `0x04` is actually correct. However, if it actually measurably improves anything, I'm fine with giving it a spin.

Reply:
take a look at RT6352_VGA_TABLE in cmm_rf_cal.c ! there is no power entry for 0x04 value causing chip to stuck at very low throughput. mtk driver maybe writes this reg on-the-fly in STA mode

Reply:
`RT6352_VGA_TABLE` in cmm_rf_cal.c isn't actually used anywhere, apart from some traces in debug messages inside DPD calibration which isn't implemented in rt2x00...

And while it's true that `RT6352_ChipAGCAdjust` got an `#ifdef CONFIG_STA_SUPPORT` around it which hints that it's only being used in STA mode, there is also `RT6352_RTMPSetAGCInitValue` which sets register 66 in the way described above independently of the mode of operation.

My guess is that 0x04 is fine for ePA and works for iPA as well if we do DPD calibration.

Reply:
in rtmp_init_inf DPD is done before RT6352_Init_ExtPA_ExtLNA. with rt2x00 LNA is always on, can't even turn it off passing 0 to TX_PIN_CFG_LNA_PE_G0_EN or TX_PIN_CFG_LNA_PE_G1_EN. calibration as such does not help, but i'm also missing a piece of code from RtmpKickOutHwNullFrame. do you know is there an equivalent in rt2x00 that would transmit a packet?